src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int SIDE_LIMIT = 7;
+    private static final int SIDE_DEFAULT = 10;
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Square square = new Square(getRandomColor(), getRandomInt(SIDE_LIMIT));
+        Rectangle rectangle =
+                new Rectangle(getRandomColor(), getRandomInt(SIDE_LIMIT), getRandomInt(SIDE_LIMIT));
+        Circle circle = new Circle(getRandomColor(), getRandomInt(SIDE_LIMIT));
+        RightTriangle rightTriangle = new RightTriangle(getRandomColor(), getRandomInt(SIDE_LIMIT));
+        IsoscelesTrapezoid trapezoid = new IsoscelesTrapezoid(getRandomColor(),
+                getRandomInt(SIDE_LIMIT), getRandomInt(SIDE_LIMIT), getRandomInt(SIDE_LIMIT));
+
+        Figure[] array = {square, rectangle, circle, rightTriangle, trapezoid};
+        int indexFigure = random.nextInt(array.length);
+
+        return array[indexFigure];
```

Comment:
lets make a switch-case 
because you are creating too much unnecessary objects

```
switch (randomNumber) {
    case 0:
         return getRandomSquare();
         ....
}
```

Reply:
Ок. I agree.